lib/src/Base/Func/InverseTrendTransform.cxx

```diff
@@ -69,24 +70,24 @@ InverseTrendTransform * InverseTrendTransform::clone() const
 String InverseTrendTransform::__repr__() const
 {
   OSS oss(true);
-  oss << "class=" << InverseTrendTransform::GetClassName()
-      << " evaluation=" << p_evaluation_->__repr__();
+  oss << "class=" << TrendTransform::GetClassName()
+      << " inherited from " << VertexValueFunction::__repr__();
   return oss;
 }
 
 /* Inverse accessor */
 TrendTransform InverseTrendTransform::getInverse() const
 {
-  if (inFld.getSpatialDimension() != p_evaluation_->getInputDimension()) throw InvalidArgumentException(HERE) << "Error: expected a Field with mesh dimension=" << p_evaluation_->getInputDimension() << ", got mesh dimension=" << inFld.getSpatialDimension();
-  if (inFld.getDimension() != p_evaluation_->getOutputDimension()) throw InvalidArgumentException(HERE) << "Error: expected a Field with dimension=" << p_evaluation_->getOutputDimension() << ", got dimension=" << inFld.getDimension();
-  ++callsNumber_;
-  return Field(inFld.getMesh(), inFld.getValues() - (*p_evaluation_)(inFld.getMesh().getVertices()));
+  return TrendTransform(getTrendFunction());
 }
 
-/* Inverse accessor */
-TrendTransform InverseTrendTransform::getInverse() const
+/* Underlying trend function accessor */
+Function InverseTrendTransform::getTrendFunction() const
 {
-  return TrendTransform(p_evaluation_);
+  // Here we know that the underlying function is based on a TrendEvaluation.
+  // Use it to recover the function defining the trend.
+  // getFunction() returns a Function which evaluation is the TrendEvaluation.
+  return static_cast<const InverseTrendEvaluation*>(getFunction().getEvaluation().get())->getFunction();
```

Comment:
this should be a dynamic cast

Reply:
@jschueller Why do I have to use a dynamic cast if I know that the cast will never fail? A static cast is both more efficient and more meaningful IMO.

Reply:
@regislebrun Unfortunately cast may fail.  For instance when using default constructor; or when load'ing a bogus Study; or maybe in other situations, who knows?